### DISCUSSION TITLE
Is it possible to specify the name of the key when response model is set to dict?

### DISCUSSION THREAD

**Discussion by ceng-p** [fastapi/fastapi#6500] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Something(BaseModel):
    a: str
    b: int
   
@app.get("/something", response_model=dict[str, Something])
async def get_something():
    some_dict = {}
    return some_dict[Something.a] = Something
```

### Description

Open the browser and navigate to /doc
In the schema for this route it will have {"additionalProp": Something}.
Is there a way to rename the additionalProp to something else on a per route basis, for example so it will say {"a": Something} instead?

### Operating System

macOS

### FastAPI Version

0.72

### Python Version

3.9.7

**Comment by hellocoldworld** [comment]
You can always[ override the openapi method](https://fastapi.tiangolo.com/advanced/extending-openapi/?h=method#override-the-method)

**Comment by ceng-p** [comment]
Closing for now as it's more to do with OpenAPI!

WIll update with my solution when I get time ✨
